File: content/atlas/source/includes/fts/field-types/autocomplete/configure_dynamic_index.cs
using MongoDB.Bson;
using MongoDB.Driver;

// connect to your Atlas deployment
var uri = "<connection-string>";

var client = new MongoClient(uri);

var db = client.GetDatabase("sample_mflix");
var collection = db.GetCollection<BsonDocument>("movies");

// define your Atlas Search index
var index =  new CreateSearchIndexModel(
  "default", new BsonDocument
  {
    { "mappings", new BsonDocument
      {
        { "dynamic", new BsonDocument { { "typeSet", "moviesStringIndex" } } },
        { "fields", new BsonDocument
          {
            { "poster", new BsonArray() },
            { "languages", new BsonArray() },
            { "rated", new BsonArray() },
            { "lastupdated", new BsonArray() },
            { "fullplot", new BsonArray() },
            { "awards", new BsonArray() }
          }
        }
      }
    },
    { "typeSets", new BsonArray
      {
        new BsonDocument
        {
          { "name", "moviesStringIndex" },
          { "types", new BsonArray
            {
              new BsonDocument { { "type", "autocomplete" } }
            }
          }
        }
      }
    }
  });

var result = collection.SearchIndexes.CreateOne(index);
Console.WriteLine($"New index name: {result}");

File: content/atlas/source/includes/fts/field-types/autocomplete/configure_dynamic_index.java
import com.mongodb.client.MongoClient;
import com.mongodb.client.MongoClients;
import com.mongodb.client.MongoCollection;
import com.mongodb.client.MongoDatabase;
import org.bson.Document;
import java.util.ArrayList;
import java.util.Arrays;

public class CreateIndex {
    public static void main(String[] args) {
        // connect to your Atlas cluster
        String uri = "<connection-string>";

        try (MongoClient mongoClient = MongoClients.create(uri)) {
            // set namespace
            MongoDatabase database = mongoClient.getDatabase("sample_mflix");
            MongoCollection<Document> collection = database.getCollection("movies");
            String indexName = "default";

            Document searchIdx = new Document(
                "mappings",
                new Document("dynamic", new Document("typeSet", "moviesStringIndex"))
                    .append("fields", new Document()
                        .append("poster", new ArrayList<>())
                        .append("languages", new ArrayList<>())
                        .append("rated", new ArrayList<>())
                        .append("lastupdated", new ArrayList<>())
                        .append("fullplot", new ArrayList<>())
                        .append("awards", new ArrayList<>())
                    )
            ).append("typeSets",
                Arrays.asList(
                    new Document("name", "moviesStringIndex")
                        .append("types", Arrays.asList(
                            new Document("type", "autocomplete")
                        ))
                )
            );

            collection.createSearchIndex(indexName, searchIdx);
            System.out.println("New index name: " + indexName);
        } catch (Exception e) {
            System.err.println("Error creating Atlas Search index: " + e.getMessage());
        }
    }
}